### PR TITLE
Adds camera zoom

### PR DESCRIPTION
**Pull request by systematiccaos**
Hi,

I love the fake-background-webcam and would like to contribute a little zoom-feature. :-)

Best

caos

### CODE REVIEW

**Comment by systematiccaos** [comment]
Added another commit to support frame-skipping for the mask calculation. No argument will update the mask every 5 frames.

**Comment by fangfufu** [comment]
You are duplicating the feature in here: 
https://github.com/fangfufu/Linux-Fake-Background-Webcam/pull/174

For each feature you add, please use a different commit. Alternatively, please amend the description of the pull request and state what you are proposing.

File: fake.py [hunk 1]
```diff
@@ -122,8 +124,11 @@ def __init__(self, args) -> None:
         self.foreground_mask = None
         self.inverted_foreground_mask = None
         self.images: Dict[str, Any] = {}
+        self.classifier = mp.solutions.selfie_segmentation.SelfieSegmentation(model_selection=1)
```
Comment (fangfufu) [inline fake.py:127]: You are calling this function again unnecessarily.

File: fake.py [hunk 2]
```diff
@@ -232,7 +236,61 @@ def compose_frame(self, frame):
...
+        # combine with the original color, oversaturated
+        out = cv2.addWeighted(img, 0.5, holo_blur, 0.6, 0)
+        return out
+
+    def paddedzoom(self, image, scale=150):
+    
+        '''
+        Zoom in/out an image while keeping the input image shape.
+        i.e., zero pad when factor<1, clip out when factor>1.
+        there is another version below (paddedzoom2)
```
Comment (fangfufu) [inline fake.py:258]: Where is your second version? If you have a second version, perhaps consider to use a parameter to switch the behaviour?

File: fake.py [hunk 3]
```diff
@@ -232,7 +236,61 @@ def compose_frame(self, frame):
...
+        '''
+
+        #get the webcam size
+        height, width, channels = image.shape
+
+        #prepare the crop
+        centerY,centerX=int(height/2),int(width/2)
+        radiusY,radiusX= int(scale*(height/2)/100),int(scale*(width/2)/100)
+        # print(f'centerX: {centerX}')
```
Comment (fangfufu) [inline fake.py:267]: Excessive amount of comments.

**Review by fangfufu — CHANGES_REQUESTED** [review]

**Comment by fangfufu** [comment]
Closing stale pull request

**Pull request closed without merging**